weneige/orders/views.py

```diff
@@ -1,3 +1,53 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.core.exceptions import ValidationError
+from django.http            import JsonResponse
+from django.views           import View
+
+from .utils.utils           import login_decorator
+from .models                import Order, OrderStatus
+from products.models        import Product, ProductImage, ProductOption, Color, Volume
+class OrderView(View):
+    @login_decorator
+    def post(self, request):
+        try :
+            data        = json.loads(request.body)
+            user        = request.user
+            product_id  = data['product_id']
+            quantity    = data['quantity']
+            color_id    = data['color_id']
+
+            Order.objects.create(
+                user            = user,
+                product_option  = ProductOption.objects.filter(product_id=product_id, color_id=color_id)[0],
+                quantity        = quantity,
+                address         = user.address,
+                mobile_number   = user.mobile_number,
+                order_status    = OrderStatus.object.get(id=1).name
+            )
+        
+            return JsonResponse({'MESSAGE' : 'SUCCESS'}, status = 201)
+        
+        except KeyError:
+            return JsonResponse({'MESSAGE' : 'KEY_ERROR'}, status = 400)
+
+    @login_decorator
+    def get(self, request):
+        user        = request.user
+        user_orders = user.user_order.filter(user_id=user.id)
+        results     = []
+
+        for user_order in user_orders:
+            results.append(
+                {
+                    "kor_name"  : Product.objects.get(id=user_order.option_order.product_id).kor_name,
+                    "eng_name"  : Product.objects.get(id=user_order.option_order.product_id).eng_name,
+                    "color"     : Color.objects.get(id=user_order.option_order.color_id).name,
+                    "volume"    : Volume.objects.get(id=user_order.option_order.color_id).name,
+                    "price"     : Product.objects.get(id=user_order.option_order.product_id).price,
+                    "quantity"  : user_order.quantity,
+                    "address"   : user_order.address,
+                    "image_url" : [url.image_url for url in ProductImage.objects.filter(id=user_order.option_order.product_id)]
+                }
+            )
+        return JsonResponse({'results' : results}, status = 200)
```

Comment:
```suggestion
@login_decorator
    def get(self, request):
        user        = request.user
        user_orders = user.user_order.filter(user_id=user.id)

        results = [{
            "kor_name"  : user_order.option_order.product.kor_name,
            "eng_name"  : user_order.option_order.product.eng_name,
            "color"     : user_order.option_order.color.name,
            "volume"    : user_order.option_order.color.name,
            "price"     : user_order.option_order.product.price,
            "quantity"  : user_order.quantity,
            "address"   : user_order.address,
            "image_url" : [url.image_url for url in ProductImage.objects.filter(id=user_order.option_order.product_id)]
            } for user_order in user_orders]
        return JsonResponse({'results' : results}, status = 200)
```

위의 피드백에 더해 list comprehension을 이용한다면 동작 속도가 더 빠른, 훨씬 더 가독성 좋은 코드를 작성 할 수 있을 것 같습니다.

리스트 컴프리헨션으로 작성한 코드는 간결하고 데이터베이스를 조회하여 리스트로 만들 때 많이 사용되므로 리스트 컴프리 헨션은 새로운 배열을 만들때 특히 유용합니다.

Reply:
40 "kor_name"  : user_order.option_order.product.kor_name 
에서 .product는 어떻게 해서 Product를 참조할 수 있는 ORM(?)으로 사용할 수 있는 건가요? option_order 처럼 제가 따로 related_name으로 설정해놓지 않은 것 같아서 그렇습니다.